manager/logbroker/broker.go

```diff
@@ -392,4 +401,13 @@ func (lb *LogBroker) PublishLogs(stream api.LogBroker_PublishLogsServer) (err er
 
 		lb.publish(log)
 	}
+
+	// check if we have any more messages, throw errors if we do, and then SendAndClose
+	_, err = stream.Recv()
```

Comment:
I'm not sure we should do this?

If the client sends a `Close` and nothing else (neither it closes the stream nor sends a message), we're going to get stuck here.

I think a better approach would just be to stop receiving when we receive `Close`